src/tasks/taskfilter.ts

```diff
@@ -10,7 +10,11 @@ export class TaskFilter {
     }
 
     public static isYieldable(task: any): task is TaskBase {
-        return (task instanceof Task) || (task instanceof TaskSet);
+        const taskBase = task as TaskBase;
+        return taskBase
+            && taskBase.isCompleted != null
+            && taskBase.isFaulted != null
+            && taskBase.yieldNewActions != null;
```

Comment:
Is the use of `null` here correct? I noticed you typically use `undefined` everywhere. Not sure what the expectation is for members that don't exist.

Reply:
`x != null` is shorthand for `x !== undefined && x !== null` due to type coalescing with the `!=` operator. I couldn't just use a falsy check here since `isCompleted` and `isFaulted` are booleans.

I think I technically could have checked just for `!== undefined` if the members don't exist.

Reply:
Good to know!